healthtools/scrapers/base_scraper.py

```diff
@@ -82,36 +85,36 @@ def scrape_site(self):
         '''
         This functions scrapes the entire website by calling each page.
         '''
-        self.set_site_pages_no()
+        try:
+            self.set_site_pages_no()
 
-        for page_num in range(1, self.site_pages_no + 1):
-            # Check if is NHIF and if so just use page_num else format site_url
-            nhif = set(re.sub(r"(\w)([A-Z])", r"\1 \2", type(self).__name__).lower().split()) &\
-                set(NHIF_SERVICES)
+            for page_num in range(1, self.site_pages_no + 1):
+                # Check if is NHIF and if so just use page_num else format site_url
+                nhif = set(re.sub(r"(\w)([A-Z])", r"\1 \2", type(self).__name__).lower().split()) &\
+                    set(NHIF_SERVICES)
 
-            url = page_num if nhif else self.site_url.format(page_num)
+                url = page_num if nhif else self.site_url.format(page_num)
 
-            try:
                 results, results_es = self.scrape_page(url, 0)
 
                 if type(results) != list:
-                    self.print_error("ERROR: scrape_site() \nsource: {} \npage: {} \ndata: {}".
+                    self.print_error("- ERROR: scrape_site() \n- SOURCE: {} \n-MESSAGE: page: {} \ndata: {}".
                                      format(url, page_num, results))
                     return
 
                 self.results.extend(results)
                 self.results_es.extend(results_es)
 
-            except Exception as err:
-                self.print_error("ERROR: scrape_site() \nsource: {} \npage: {} \nerr: {}".format(url, page_num, err))
-                return
+            if self.results:
+                self.archive_data(json.dumps(self.results))
+                self.elasticsearch_delete_docs()
+                self.elasticsearch_index(self.results_es)
 
-        if self.results:
-            self.archive_data(json.dumps(self.results))
-            self.elasticsearch_delete_docs()
-            self.elasticsearch_index(self.results_es)
+                return self.results
 
-            return self.results
+        except Exception as err:
+            self.print_error("- ERROR: scrape_site() \n- SOURCE: {} \n- MESSAGE: {}".format(self.site_url, err))
+            return
```

Comment:
We should probably have the try-catch statement in the `set_site_pages_no` function instead and check the return value when we call the function at line 85/89 above.